src/main/java/Ui.java

```diff
@@ -0,0 +1,84 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * Handles the interactions with the user as they input to Duke.
+ */
+public class Ui {
```

Comment:
I like your `Ui` class and how you've abstracted a lot of logic/code related to printing to behind all the various methods. One thing I noticed - your `Ui` class seems to be missing JavaDocs for all of its methods. Might be tedious but just a heads-up for you!